### PR TITLE
refactor: re-export functions from index.d.ts

### LINKED ISSUE #120: Re-Export functions instead of declare again
**What would you like to be able to do?**

All declared functions in `index.d.ts` should be removed and replaced by the relevant exports from lib.

**Did you already have any implementation ideas?**
yes

/cc @RahulGautamSingh

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes:

- Re-export functions from `index.d.ts` instead of declaring again.

## Context:

- closes #120

### CODE REVIEW

File: lib/semantic.d.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import type { SemVer } from "semver";
 
+export function inc(input: string | SemVer): string | null;
 export function major(input: string | SemVer): number;
 export function minor(input: string | SemVer): number;
 export function patch(input: string | SemVer): number;
```
Comment (viceice) [inline lib/semantic.d.ts:6]:
```suggestion
export function inc(input: string): string | null;
export function major(input: string): number;
export function minor(input: string): number;
export function patch(input: string): number;
```

It seems we don't support `SemVer` class in js code, so remove it.

File: lib/specifier.d.ts [hunk 2]
```diff
@@ -4,12 +4,12 @@ export const RANGE_PATTERN: string;
 
 /*
```
Comment (viceice) [inline lib/specifier.d.ts:5]: please remove this comment

File: lib/operator.d.ts [hunk 3]
```diff
@@ -1,9 +1,9 @@
+export function arbitrary(version: string, other: string): boolean;
```
Comment (viceice) [inline lib/operator.d.ts:1]:
```suggestion
```
not exported

https://github.com/renovatebot/pep440/blob/5fb26d3568620c980f92a09d7e32561cae314613/lib/operator.js#L3-L19

Comment (viceice) [reply lib/operator.d.ts:1]: We can ignore the string exports, as they can't really imported.

Comment (RahulGautamSingh) [reply lib/operator.d.ts]: There are tests which require the string operators.
So leaving them as exported for now.

Comment (RahulGautamSingh) [reply lib/operator.d.ts]: We need those string operator exports for tests.

**Review by viceice — APPROVED** [review]

**Pull request merged**